copier/template.py

```diff
@@ -443,6 +443,10 @@ def local_abspath(self) -> Path:
             raise ValueError("Local template must be a directory.")
         return result.absolute()
 
+    @cached_property
+    def is_temp_clone(self) -> bool:
+        return self.local_abspath != Path(self.url).absolute()
```

Comment:
I think you can just have a normal property called `_temp_clone`, which is filled up with the clone path inside `local_abspath()`, only if required.

Then add `_cleanup()` and remove that path recursively if it exists, just like you were already doing.